tests/load/locustfiles/locustfile.py

```diff
@@ -131,19 +172,56 @@ def register(self) -> None:
             AssertionError: If the user fails to register a channel
             ValidationError: If the register message schema is not as expected
         """
+        if not self.uaid:
+            raise RescheduleTask()
+
         chid: str = str(uuid.uuid4())
+        control: bool = True
+
+        with self._time_event(name="send_notification") as timer:
+            while control:
+                try:
+                    body = json.dumps(dict(messageType="register", channelID=chid))
+                    self.ws.send(body)
+                    reply = self.ws.recv()
+                except ssl.SSLError:
+                    # if there is an error, disconnect and retry
+                    self.disconnect()
+                    self.connect()
+                    self.hello()
+                else:
+                    res: RegisterMessage = RegisterMessage(**json.loads(reply))
+                    assert (
+                        res.status == 200
+                    ), f"Unexpected status. Expected: 200 Actual: {res.status}"
+                    assert (
+                        res.channelID == chid
+                    ), f"Channel ID did not match, received {res.channelID}"
+                    timer.response_length = len(reply.encode("utf-8"))
+                    self.channels[chid] = res.pushEndpoint
+                    control = False
+
+    @task(weight=95)
+    def send_notification(self):
+        """
+        Sends a notification to a registered enpoint
 
-        with self._time_event(name="register") as timer:
-            body = json.dumps(dict(messageType="register", channelID=chid))
-            self.ws.send(body)
-            reply = self.ws.recv()
-            assert reply, f"No 'register' response CHID: {chid}"
-            res: RegisterMessage = RegisterMessage(**json.loads(reply))
-            assert res.status == 200, f"Unexpected status. Expected: 200 Actual: {res.status}"
-            assert res.channelID == chid, f"Channel ID did not match, received {res.channelID}"
-            timer.response_length = len(reply.encode("utf-8"))
-        self.channels[chid] = res.pushEndpoint
+        Raises:
+            AssertionError: If the server does not respond correctly (400, 500, etc)
+        """
+        if not self.channels:
+            raise RescheduleTask()
 
-    @task
-    def do_nothing(self) -> None:
-        pass
+        self.disconnect()
+
+        with self._time_event(name="send_notification") as timer:
+            endpoint_url = random.choice(list(self.channels.items()))[-1]
+            endpoint_res = self.client.post(
+                url=endpoint_url,
+                name="Endpoint Notification",
+                data=self.encrypted_data,
+                headers=self.headers,
+            )
+            assert endpoint_res.status_code == 201, f"status code was {endpoint_res.status_code}"
```

Comment:
Because we are using the `self.client` an assertion isn't the appropriate choice for verification. Instead we change the response status. Something like:

```suggestion
            if response.status_code != 201:
                response.failure(f"{response.status_code=}, expected 201, {response.text=}")
                return
```